packages/loader/container-loader/src/container.ts

```diff
@@ -975,7 +978,12 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             // Note: no need to fetch ops as we do it preemptively as part of DeltaManager.attachOpHandler().
             // If there is gap, we will learn about it once connected, but the gap should be small (if any),
             // assuming that connect() is called quickly after initial container boot.
-            this.connectInternal({ reason: "DocumentConnect", fetchOpsFromStorage: false });
+            this._connectAbortController = new AbortController();  // Create a new signal for each connect() call
+            this.connectInternal({
+                reason: "DocumentConnect",
+                fetchOpsFromStorage: false,
+                progress: { cancel: this._connectAbortController.signal },
```

Comment:
Side comment, FYI @vladsud too.

IProgress has this comment on `onRetry`:

> Any exception thrown from this call back result in cancellation of operation and propagation of thrown exception.

This is a smell that as-is `IProgress` is tightly coupled to `runWithRetry` in `driver-utils` package, which is where that logic is actually implemented.  No way to enforce that behavior at the interface level, of course.  We might want to untangle that soon, and maybe even not use `IProgress` until this is cleared up. e.g. switch it to a class that includes `runWithRetry` or something like it (not sure what shape we'd need for this PR!) Or maybe the comment just needs to be updated to say "It's recommended that" in the front.